lumigator/backend/backend/services/workflows.py

```diff
@@ -120,104 +194,19 @@ def create_workflow(
-        # have implemented the association of workflows with experiments
-        workflow_record = {
-            "id": uuid.uuid4(),
-            "experiment_id": request.experiment_id,
-            "name": request.name,
-            "description": request.description,
-            "created_at": created_at,
-            "updated_at": created_at,
-        }
-
-        # TODO: This part will need to be refactored more:
-        # once all the jobs are done, the last
-        # step is to created two jobs inside the workflow_record
-        # which store the inference and evaluation job output info
-        # on_job_complete_store_in_tracking_service()....
-        workflow_record["status"] = JobStatus.CREATED
-
-        return WorkflowResponse.model_validate(workflow_record)
-
-    # TODO: until we have implemented the association of workflows with experiments,
-    #  everything continues to be indexed by experiment_id
-    def get_workflow_jobs(self, experiment_id: UUID) -> ListingResponse[JobResponse]:
-        records = self._job_repo.get_by_experiment_id(experiment_id)
-        return ListingResponse(
-            total=len(records),
-            items=[JobResponse.model_validate(x) for x in records],
-        )
+        # Run the inference and evaluation pipeline as a background task
+        background_tasks.add_task(self._run_inference_eval_pipeline, workflow, request)
 
-    def get_workflow_result_download(self, experiment_id: UUID) -> WorkflowResultDownloadResponse:
-        """Return experiment results file URL for downloading."""
-        s3 = S3FileSystem()
-        # get jobs matching this experiment
-        # have a query returning a list of (two) jobs, one inference and one eval,
-        # matching the current experiment id. Note that each job has its own type baked in
-        # (per https://github.com/mozilla-ai/lumigator/pull/576)
-        jobs = self.get_workflow_jobs(experiment_id)
+        return workflow
 
-        # iterate on jobs and depending on which job this is, import selected fields
-        results = {}
-
-        for job in jobs:
-            # Get the dataset from the S3 bucket
-            result_key = self._job_service._get_results_s3_key(job.id)
-            with s3.open(f"{settings.S3_BUCKET}/{result_key}", "r") as f:
-                job_results = json.loads(f.read())
-
-            # we just merge the two dictionaries for now
-            results.update(job_results)
-
-        loguru.logger.error(results)
-
-        # Generate presigned download URL for the object
-        result_key = self._job_service._get_results_s3_key(experiment_id)
-        download_url = self._dataset_service.s3_client.generate_presigned_url(
-            "get_object",
-            Params={
-                "Bucket": settings.S3_BUCKET,
-                "Key": result_key,
-            },
-            ExpiresIn=settings.S3_URL_EXPIRATION,
-        )
+    def delete_workflow(self, workflow_id: str) -> WorkflowResponse:
```

Comment:
I assume that this would mean "delete workflow run" until we have a proper workflow record associated with the experiment in which it's included, and with the jobs it contains, right?

Reply:
I think we chatted about this in another comment, but I'd like to steer away from using `run` as a noun, since it's an mlflow specific term. the tracking_server api is creating and deleting workflows, while the backend service is running the workflows and checking their status. 

In this situation, this delete_worklfow function deletes from mlflow the workflow and any jobs associated with it. However, I don't handle the case where a user tries to delete a workflow that is still running. I'll update this code to prevent the deletion of a workflow that is in progress (aka has jobs attached to it that haven't yet completed) Thanks!